services/121-service/src/registration/registrations.service.ts

```diff
@@ -760,24 +762,77 @@ export class RegistrationsService {
       );
 
       if (message) {
-        this.sendSmsMessage(registration, programId, message);
+        this.sendTextMessage(registration, programId, message);
       }
     }
   }
 
-  private async sendSmsMessage(
+  private async sendTextMessage(
     registration: RegistrationEntity,
     programId: number,
     message?: string,
+    key?: string,
   ): Promise<void> {
-    this.smsService.notifyBySms(
-      registration.id,
-      registration.phoneNumber,
-      registration.preferredLanguage,
-      programId,
-      message,
-      null,
-    );
+    if (!message && !key) {
+      throw new HttpException(
+        'A message or a key should be supplied.',
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    let whatsappNumber;
+    if (registration.customData) {
+      whatsappNumber =
+        registration.customData[CustomDataAttributes.whatsappPhoneNumber];
+    }
+    if (registration.phoneNumber && !whatsappNumber) {
```

Comment:
@RubenGeo what is this? Doesn't this stop the flow in case no whatsappNumber is found (while it should continue to send am SMS?)

Reply:
@RubenGeo I now tested this as well, and this indeed goes wrong. I think there should be a ! also in front of registration.phoneNumber right?